tutorials/embedded-c-getting-started/index.md

```diff
@@ -0,0 +1,313 @@
+char *publish_topic = NULL;
+
+asprintf(&publish_topic, PUBLISH_TOPIC_EVENT, CONFIG_GIOT_DEVICE_ID);
+
+char *publish_message = NULL; asprintf(&publish_message, TEMPERATURE_DATA, MIN_TEMP + rand() % 10);
+
+ESP_LOGI(TAG, "publishing msg \"%s\" to topic: \"%s\"\n", publish_message, publish_topic);
+
+iotc_publish(context_handle, publish_topic, publish_message, iotc_example_qos,NULL,NULL);
+free(publish_topic); free(publish_message);
+}
+```
+
+## Sending Commands from Cloud IoT Core to ESP32
+The callback function is invoked when the device receives a message from the cloud. This is where the code turns the led on and off based on incoming messages.You can find this code in the mqtt-example.c file on line 50.
+
+```c
+void iotc_mqttlogic_subscribe_callback(iotc_context_handle_t in_context_handle, iotc_sub_call_type_t call_type, const iotc_sub_call_params_t *const params, iotc_state_t state, void *user_data) {
+
+    char *sub_message = (char *)malloc(params->message.temporary_payload_data_length + 1);
+
+    memcpy(sub_message, params->message.temporary_payload_data, params->message.temporary_payload_data_length);
+
+    sub_message[params->message.temporary_payload_data_length] = '\0';
+
+    ESP_LOGI(TAG, "Delegate Message Payload: %s", sub_message);
+
+    if (strcmp(subscribe_topic_command, params->message.topic) == 0)
+    {
+        gpio_pad_select_gpio(BLINK_GPIO);
+        /* Set the GPIO as a push/pull output */
+        gpio_set_direction(BLINK_GPIO, GPIO_MODE_OUTPUT);
+
+        int value;
+        sscanf(sub_message, "light: %d", &value);
+        ESP_LOGI(TAG, "value: %d\n", value);
+        if (value == 1)
+        {
+            ESP_LOGI(TAG, "ON");
+            gpio_set_level(BLINK_GPIO, 1);
+        }
+        else if (value == 0)
+        {
+            gpio_set_level(BLINK_GPIO, 0);
+        }
+    }
+    free(sub_message);
+}
+```
+
+## Running the Sample
+
+**To Connect to Cloud:**
+
+1. Use menu configuration with `make`
+
+```c
+cd /examples/main/ make menuconfig
+```
+
+1. Set up your WiFi and LED gpio pin , navigate to example configuration
+
+![Wifi Setup](https://github.com/galz10/community/blob/gal/tutorials/embedded-c-getting-started/wifisetup.gif)
+
+1. Set up your Google Cloud Project information, navigate to component configuration and then to IoT Core Configuration
+
+![Cloud Project Setup](https://github.com/galz10/community/blob/gal/tutorials/embedded-c-getting-started/CloudSetup.gif)
+
+1. Locate your ec_private.pem file and copy its contents into the private.pem file in the certs folder located at examples/main/certs
+1. Run the following to build your program Idf.py build inside the examples folder
+1. Run the following to flash the program onto your device, (port) is the port you’re using. Idf.py -p /dev/cu.usbserial1440 flash
+1. Run the following to monitor your device Idf.py -p /dev/cu.usbserial1440 monitor
```

Comment:
Mabe better as:

1. Run `idf.py build` to build sources into firmware
2. Run `idf.py -p /dev/cu.usbserial1440 monitor` passing the path to your tty device to monitor the device output.